pontoon/insights/models.py

```diff
@@ -0,0 +1,47 @@
+from django.contrib.auth.models import User
+from django.db import models
+from django.utils import timezone
+
+from pontoon.base.models import AggregatedStats
+
+
+class InsightsSnapshot(AggregatedStats, models.Model):
+    created_at = models.DateField(default=timezone.now)
+
+    # Active users
+    active_managers = models.ManyToManyField(
+        User, related_name="active_managers_insights"
+    )
+    all_managers = models.ManyToManyField(User, related_name="all_managers_insights")
+    active_reviewers = models.ManyToManyField(
+        User, related_name="active_reviewers_insights"
+    )
+    all_reviewers = models.ManyToManyField(User, related_name="all_reviewers_insights")
+    active_contributors = models.ManyToManyField(
+        User, related_name="active_contributors_insights"
+    )
+    all_contributors = models.ManyToManyField(
+        User, related_name="all_contributors_insights"
+    )
```

Comment:
Active users data is different from all the other data in that we can't simply aggregate daily numbers and use them as the input for the charts. We need to observe the entire period (in our case 12 months) and count how many different users did we have in each of the observed groups.

Hence the use of the `many-to-many` relationship with the User model. Which is tedious, because we're saving bug amounts of data in order to get to 6 numbers eventually and because it adds complexity to `get_insights()`.

I'm sort of hoping to be able remove these fields completely as part of [bug 1680054](https://bugzilla.mozilla.org/show_bug.cgi?id=1680054) if it turns out we can calculate the numbers on the fly (on each dashboard load) quickly enough.

Reply:
> I'm sort of hoping to be able remove these fields completely as part of [bug 1680054](https://bugzilla.mozilla.org/show_bug.cgi?id=1680054) if it turns out we can calculate the numbers on the fly (on each dashboard load) quickly enough.

It turns out we should still keep the numbers in the DB, but I was able to greatly simplify the data model by storing active user numbers for various periods separately.